atlascope/core/models/dataset_embedding.py

```diff
@@ -0,0 +1,42 @@
+from uuid import uuid4
+
+from django.contrib import admin
+from django.contrib.gis.db.models import PointField
+from django.db import models
+from rest_framework import serializers
+
+
+class DatasetEmbedding(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid4, editable=False)
+    context = models.ForeignKey('Investigation', on_delete=models.CASCADE)
+    parent = models.ForeignKey(
+        'Dataset',
+        on_delete=models.CASCADE,
+        related_name='child_embeddings',
+    )
+    child = models.ForeignKey(
+        'Dataset',
+        on_delete=models.CASCADE,
+        related_name='parent_embeddings',
+    )
+    child_upper_left = PointField()
+    child_lower_right = PointField()
```

Comment:
Can we combine these into a single bounding box? It will likely result in easier to write and more performant queries.

Reply:
Perhaps we could enforce that the coordinates are indeed "`upper_left`" relative to "`lower_right`" in the meantime?

Reply:
I think a change ought to be made before merging. Whichever is easier: the constraints or bounding box.

Reply:
Switched to using a Polygon from bounding box in [8d2e8d5](https://github.com/atlascope/atlascope/pull/82/commits/8d2e8d590ed2a15e0274b50163ce62d9f1d91668). I'm thinking we can add a db constraint later when we're sure what constraints apply (e.g. should we allow shapes other than rectangles?)

Reply:
> I'm thinking we can add a db constraint later when we're sure what constraints apply

Totally agree. This is what I need to move forward on the frontend.

> should we allow shapes other than rectangles?

Probably not, but I don't know yet. I will need to assume these polygons are rectilinear in order to prevent blowing up the complexity of the frontend. The frontend will break when this assumption breaks. We'll probably need to add that constraint in sooner-or-later. I'll open an issue after this gets merged.

Reply:
I know this is merged already, but there's no reason to worry about anything besides rectangles right now, especially if the rectangle assumption keeps things easy to work on for the rest of Phase 1.